extensions/arc/deployment/src/test/java/io/quarkus/arc/test/config/ConfigDefaultValuesTest.java

```diff
@@ -0,0 +1,112 @@
+package io.quarkus.arc.test.config;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.arc.deployment.ConfigPropertyBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class ConfigDefaultValuesTest {
+    @RegisterExtension
+    static final QuarkusUnitTest TEST = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClass(ConfigDefaultValuesTest.class)
+                    .addClass(MapBuildTimeConfig.class)
+                    .addClass(MapBuildTimeConfigBuildStep.class)
+                    .addAsResource(new StringAsset(MapBuildTimeConfig.class.getName()), "META-INF/quarkus-config-roots.list")
```

Comment:
This is not something that is supported, it just happens to work. There are lots of things that could go wrong here (e.g. there are other copies of this file on the class path, it only picks up the correct one by chance).

Can you move this test to core/test-extension and put the config classes in the extension itself? Its where all the other config based tests are so I think it is where it really belongs.

Reply:
Interesting. I've got the idea that all the files from the classpath would load and it was fine.

No worries. Let me change that. Thanks!